src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -479,35 +486,91 @@ private void Load()
             Receive(new Action<Write>(IgnoreDebug));
             Receive(new Action<Status>(IgnoreDebug));
             Receive(new Action<Gossip>(IgnoreDebug));
+
+            Receive<ClusterEvent.IClusterDomainEvent>(msg =>
+            {
+                if(!NormalReceive(msg))
+                    Unhandled(msg);
+            });
+
+            ReceiveAny(_ => Stash.Stash());
         }
 
-        private void NormalReceive()
+        private bool NormalReceive(object message)
```

Comment:
This method is practically the same, needed to change it to untyped receive so that we can call it from somewhere else and checks if it handled the message or not (partial function in scala)

Reply:
And that works even with a `ReceiveActor` ? Never tried that before.

Reply:
Good point, I should probably change the base class to `UntypedActor` instead.

Reply:
Sounds good to me